.github/workflows/clang-format.yml

```diff
@@ -0,0 +1,41 @@
+name: clang-format-check
```

Comment:
Tested this by creating new repo and using this action with llvm style flagged to be formatted into Google style.
Verified works only with the differences in a file and not entire file

Reply:
Test Example:
Added ill-formatted cpp code
<img width="839" alt="Screenshot 2024-11-01 at 11 46 15 AM" src="https://github.com/user-attachments/assets/571aa8fa-f0c6-49be-9ba1-e106e318e6a9">

Reply:
Github actions catch formatting issues as expected on the PR

<img width="1254" alt="Screenshot 2024-11-01 at 11 46 46 AM" src="https://github.com/user-attachments/assets/eaaf4788-e79a-4deb-92d0-ce410f5fecba">